Assets/Scripts/GunBody.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GunBody : MonoBehaviour
+{
+    [SerializeField]
+    private GunStats stats;
+
+    // Base stats of the gun
+    public GunStats Stats { get => Instantiate(stats); }
```

Comment:
Maybe I've misunderstood something here, but I think It'd be ideal to only instantiate it once?

```cs
// template scriptableobject
[SerializeField]
private GunStats stats;

// Base stats of the gun
private GunStats statsInstance;
public GunStats Stats 
{ 
    get 
    {
        if (statsInstance == null)
            statsInstance = Instantiate(stats); 
        return statsInstance;
    }
}

Reply:
Or, if not: I think it should be explicitised that we're making a new instance. For example by changing the getter-property to a function aptly named to clarify the instantiation.

Reply:
The gunbody Stats will usually only be called once and passed as an instance to guncontroller. Whenever we remake the gun, it will be called again, reseting the stats, which is intentional. However this could be trouble is someone decides to get the stat directly from the body.

Reply:
I'm slightly concerned, but this should be fine as long as we don't use it wrongly